pkg/cache/tas_flavor_snapshot.go

```diff
@@ -296,6 +297,25 @@ func levelKey(topologyRequest *kueue.PodSetTopologyRequest) *string {
 	return nil
 }
 
+// findBestFitDomain binsearches a sorted array of domains and finds a domain
+// with the lowest value of state, higher or equal than count.
+// If such a domain doesn't exist, it returns the first element of the array
+func findBestFitDomain(domains []*domain, count int32) *domain {
```

Comment:
how many domains do we expect at most - and does this number justify using binary search versus linear search? if yes, can we reuse https://go.dev/src/sort/search.go?

Reply:
I believe we should support clusters that can have dozens of thousands of nodes so this is the number of domains we should support. Thanks for pointing to the `search.go`, I'm on it

Reply:
Done, ptal